### ISSUE TITLE
"kbox destroy" leaves files behind

### ISSUE THREAD

**Issue by ptmkenny** [kalabox/kalabox#969]
kbox 0.11.1
"kbox destroy" appears to leave behind the site code and various other directories.

I created a box for a Drupal 7 site through `kbox create pantheon`, then ran `kbox mybox destroy`, and the result is the following output:
```

17:dev pk$ kbox npp destroy
###############################################################

             WARNING! APP DESTRUCTION IMMINENT

  Before you continue please be aware that after you destroy
  an app there will be nothing left. That means yours app's
  codes, files, databases, indeces, caches, containers and
  EVERYTHING ELSE that has to do with your app WILL BE GONE.

###############################################################
? So, are you still prepared to DESTROY? Yes
info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/Users/pk/.kalabox/downloads/kalabox/kalabox-1.yml, 4=ps, 5=-q, 6=syncthing, 7=undefined
info: COMPOSE =>  0=--project-name, 1=npp, 2=--file, 3=/Users/pk/dev/npp/kalabox-compose.yml, 4=--file, 5=/Users/pk/dev/npp/kalabox-cli.yml, 6=stop, mode=collect
info: DOCKER-COMPOSE =>  Stopping npp_edge_1 ...
Stopping npp_appserver_1 ...
Stopping npp_appserver_1 ... done
Stopping npp_edge_1 ... done
Stopping npp_redis_1 ... done
Stopping npp_solr_1 ... done
Stopping npp_db_1 ... done
info: DOCKER-COMPOSE =>
info: DOCKER-COMPOSE =>
info: DOCKER-COMPOSE =>
info: DOCKER-COMPOSE =>  Run exited with code: 0
info: COMPOSE =>  0=--project-name, 1=npp, 2=--file, 3=/Users/pk/dev/npp/kalabox-compose.yml, 4=--file, 5=/Users/pk/dev/npp/kalabox-cli.yml, 6=rm, 7=--force, 8=-v, mode=collect
info: DOCKER-COMPOSE =>  Going to remove npp_terminus_1, npp_cli_1, npp_edge_1, npp_appserver_1, npp_db_1, npp_solr_1, npp_data_1, npp_redis_1
Removing npp_terminus_1 ... done
Removing npp_cli_1 ...
Removing npp_edge_1 ...
Removing npp_appserver_1 ...
Removing npp_db_1 ...
Removing npp_edge_1 ... done
Removing npp_data_1 ...
Removing npp_cli_1 ... done
info: DOCKER-COMPOSE =>
info: DOCKER-COMPOSE =>
Removing npp_solr_1 ... done
info: DOCKER-COMPOSE =>
info: DOCKER-COMPOSE =>
Removing npp_redis_1 ... done
info: DOCKER-COMPOSE =>
info: DOCKER-COMPOSE =>  Run exited with code: 0
info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/Users/pk/.kalabox/downloads/kalabox/kalabox-2.yml, 4=run, 5=--entrypoint, 6=rm, 7=--rm, 8=syncthing, 9=-rf, 10=/code/npp, mode=false
###############################################################

                 DESTRUCTION COMPLETE

###############################################################
```
**But when I cd into the directory where I installed the box, there is a bunch of stuff left, including the code:**
```


-rw-rw-r--   1 pk  staff   3.2K Feb  5 13:56 cli.yml
drwxrwxr-x  34 pk  staff   1.1K Feb  5 14:01 code
drwxrwxr-x  11 pk  staff   374B Feb  5 13:56 config
drwxrwxr-x   8 pk  staff   272B Feb  5 13:56 dockerfiles
-rw-rw-r--   1 pk  staff   3.4K Feb  5 13:56 kalabox-cli.yml
-rw-rw-r--   1 pk  staff   6.7K Feb  5 13:56 kalabox-compose.yml
-rw-rw-r--   1 pk  staff   640B Feb  5 13:56 kalabox.yml
drwxrwxr-x  40 pk  staff   1.3K Feb  5 13:56 node_modules
-rw-rw-r--   1 pk  staff   168B Feb  5 13:56 package.json
drwxrwxr-x   3 pk  staff   102B Feb  5 13:56 plugins
``

**Comment by pirog** [comment]
This is by design. The message should be updated.

On Thursday, February 4, 2016, ptmkenny <notifications@github.com> wrote:

> kbox 0111
> "kbox destroy" appears to leave behind the site code and various other
> directories
>
> I created a box for a Drupal 7 site through kbox create pantheon, then
> ran kbox mybox destroy, and the result is the following output:
>
>
> 17:dev pk$ kbox npp destroy
> ###############################################################
>
>              WARNING! APP DESTRUCTION IMMINENT
>
>   Before you continue please be aware that after you destroy
>   an app there will be nothing left That means yours app's
>   codes, files, databases, indeces, caches, containers and
>   EVERYTHING ELSE that has to do with your app WILL BE GONE
>
> ###############################################################
> ? So, are you still prepared to DESTROY? Yes
> info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/Users/pk/kalabox/downloads/kalabox/kalabox-1yml, 4=ps, 5=-q, 6=syncthing, 7=undefined
> info: COMPOSE =>  0=--project-name, 1=npp, 2=--file, 3=/Users/pk/dev/npp/kalabox-composeyml, 4=--file, 5=/Users/pk/dev/npp/kalabox-cliyml, 6=stop, mode=collect
> info: DOCKER-COMPOSE =>  Stopping npp_edge_1
> Stopping npp_appserver_1
> Stopping npp_appserver_1  done
> Stopping npp_edge_1  done
> Stopping npp_redis_1  done
> Stopping npp_solr_1  done
> Stopping npp_db_1  done
> info: DOCKER-COMPOSE =>
> info: DOCKER-COMPOSE =>
> info: DOCKER-COMPOSE =>
> info: DOCKER-COMPOSE =>  Run exited with code: 0
> info: COMPOSE =>  0=--project-name, 1=npp, 2=--file, 3=/Users/pk/dev/npp/kalabox-composeyml, 4=--file, 5=/Users/pk/dev/npp/kalabox-cliyml, 6=rm, 7=--force, 8=-v, mode=collect
> info: DOCKER-COMPOSE =>  Going to remove npp_terminus_1, npp_cli_1, npp_edge_1, npp_appserver_1, npp_db_1, npp_solr_1, npp_data_1, npp_redis_1
> Removing npp_terminus_1  done
> Removing npp_cli_1
> Removing npp_edge_1
> Removing npp_appserver_1
> Removing npp_db_1
> Removing npp_edge_1  done
> Removing npp_data_1
> Removing npp_cli_1  done
> info: DOCKER-COMPOSE =>
> info: DOCKER-COMPOSE =>
> Removing npp_solr_1  done
> info: DOCKER-COMPOSE =>
> info: DOCKER-COMPOSE =>
> Removing npp_redis_1  done
> info: DOCKER-COMPOSE =>
> info: DOCKER-COMPOSE =>  Run exited with code: 0
> info: COMPOSE =>  0=--project-name, 1=kalabox, 2=--file, 3=/Users/pk/kalabox/downloads/kalabox/kalabox-2yml, 4=run, 5=--entrypoint, 6=rm, 7=--rm, 8=syncthing, 9=-rf, 10=/code/npp, mode=false
> ###############################################################
>
>                  DESTRUCTION COMPLETE
>
> ###############################################################
>
> **
> But when I cd into the directory where I installed the box, there is a
> bunch of stuff left, including the code:**
>
>
>
> -rw-rw-r--   1 pk  staff   32K Feb  5 13:56 cliyml
> drwxrwxr-x  34 pk  staff   11K Feb  5 14:01 code
> drwxrwxr-x  11 pk  staff   374B Feb  5 13:56 config
> drwxrwxr-x   8 pk  staff   272B Feb  5 13:56 dockerfiles
> -rw-rw-r--   1 pk  staff   34K Feb  5 13:56 kalabox-cliyml
> -rw-rw-r--   1 pk  staff   67K Feb  5 13:56 kalabox-composeyml
> -rw-rw-r--   1 pk  staff   640B Feb  5 13:56 kalaboxyml
> drwxrwxr-x  40 pk  staff   13K Feb  5 13:56 node_modules
> -rw-rw-r--   1 pk  staff   168B Feb  5 13:56 packagejson
> drwxrwxr-x   3 pk  staff   102B Feb  5 13:56 plugins
> ``
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/969>.
>


-- 
Mike Pirog
Co-Founder, Kalabox Inc

**Comment by labboy0276** [comment]
Yes, destroy removes the docker containers only.  It is usually a good thing to leave the files when you have custom yml files / config or a non standard app.  Once they are gone, they are gone and you may have not saved the config.

@pirog any thoughts to add an option to destroy the files post install?  99.1% of the time I do a cd .. && rm-rf at the end of kbox destroy.

**Comment by pirog** [comment]
So i think we have to leave this part up to the user for various reasons. Primarily among them, if you have "protected" files like settings.php or some config and they are not owned correctly by the host machine then we are not going to be able to delete them. 

I suppose we could try to delete what we can and leave the rest for the user to handle but that seems potentially dark and full of terror.

**Comment by alphex** [comment]
as a UX experience, there should be clarification on what it does.  and giving the options or interface to simply "disable the app"  or "destroy" (completely remove EVERYTHING).

As another use case / consequence.

When I installed the GUI, I pulled down a D7 site, and named it "test" in kalabox, that gave me "test.kbox" in the browser.

Now, for reasons I couldn't determine (so I didn't open an issue),  that kalabox site never pulled down anything in to the CODE directory.   

I deleted the site in kalabox, and tried to "add a site" in kalabox (the same site).
I used the same site name.
test.

This also had the same problem.

I made ANOTHER site in kalabox, and gave it a new name, and everything worked at that point.

... Having the UX in place to destroy, is important, so that people know what they're doing with their sites and apps.   

This is more of a feature request, then a bug.   

Thanks.
